src/runtime/lpython_parser.py

```diff
@@ -35,6 +35,8 @@ def visit_Constant(self, node):
                         node.value.imag, node.kind)
         elif isinstance(node.value, dict):
             return python_ast.Dict(node.keys, node.values)
+        elif isinstance(node.value, tuple):
+            return python_ast.Tuple(node.elts, node.ctx)
```

Comment:
The other thing I noticed is that I think you have to convert "node.elts" from Python's AST to `python_ast` (our module). Otherwise they still refer to Python's AST, which does not know how to serialize, so it should fail... I don't understand why it does not fail. Perhaps because the types are simple. 

Try constructing a tuple of tuple.

Reply:
```python
a = ((1, 2), (3, 4))
```
The above code leaves us with:
```bash
LFortranException: Not implemented
``` 
because it fails to get the type when it hits this line:
```c++
tuple_type_vec.push_back(al, ASRUtils::expr_type(expr))
```
So it works because the types are simple.

Reply:
The code was not being executed, so we removed it.